firefox-ios/Client/Frontend/Browser/RecordVisitObservationManager.swift

```diff
@@ -0,0 +1,66 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Common
+import MozillaAppServices
+import Shared
+import Storage
+import WebKit
+
+// Handles recording visits to websites, which will be displayed in the History Panel.
+class RecordVisitObservationManager {
+    private var historyHandler: HistoryHandler
+    nonisolated let logger: Logger
+    var lastObservationRecorded: VisitObservation?
+
+    init (historyHandler: HistoryHandler,
+          logger: Logger = DefaultLogger.shared) {
```

Comment:
There was an extra space here?
```suggestion
    init(historyHandler: HistoryHandler,
         logger: Logger = DefaultLogger.shared) {
```